plugins/paloalto_wildfire/komand_paloalto_wildfire/actions/submit_file_from_url/action.py

```diff
@@ -33,35 +34,26 @@ def run(self, params={}):
         # 00da: pdf
         # 00fd: --------------------------8557684369749613--
         # ...
-
-        req = {
-            "apikey": (None, self.connection.api_key),
-            "url": (None, params.get("url")),
-        }
-
-        try:
-            r = requests.post(url, files=req)
-            o = xmltodict.parse(r.content)
-            out = dict(o["wildfire"]["upload-file-info"])
-        except:
-            self.logger.info("Error occurred")
-            raise
-
-        if not out["filename"]:
-            out["filename"] = "Unknown"
-
-        return {"submission": out}
-
-    def test(self):
-        """TODO: Test action"""
-        client = self.connection.client
-        return {
-            "submission": {
-                "filetype": "Test",
-                "filename": "Test",
-                "url": "Test",
-                "sha256": "Test",
-                "md5": "Test",
-                "size": "Test",
-            }
-        }
+        url = params.get(Input.URL)
+        if Utils.check_link_for_supported_file_type(url):
+            file_from_url = self.connection.client.get_file_from_url(url=url)
+            verdict = self.connection.client.get_verdicts(analysed_hash=hashlib.sha256(file_from_url).hexdigest())
+            if verdict == UNKNOWN_VERDICT:
+                try:
+                    o = xmltodict.parse(self.connection.client.submit_file_from_url(url))
+                    out = dict(o["wildfire"]["upload-file-info"])
+                except PluginException:
+                    self.logger.info("Error occurred")
+                    raise
```

Comment:
Can we also add an except for `KeyError` here in case if these keys `["wildfire"]["upload-file-info"]` are not returned in the object?